src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1509,196 @@ public void addUpdateRemoveConditionalFormat() throws Exception
         assertEquals(true, editedAquamap.get("bold"));
     }
 
+    @Test
+    public void testExportImportListFields() throws Exception
+    {
+        String listName = "exportFieldsTestList";
+
+        // create a list
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        List<PropertyDescriptor> fields = importExportFields();
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(fields);
+        // collect the fields from the server response
+        List<PropertyDescriptor> createdFields = dgen.createList(createDefaultConnection(), "Key")
+                .getDomain().getFields();
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldsPanel();
+
+        // export the fields via the UI
+        File exportFile = domainFormPanel.clickExportFields();
+        List<String> uiFields = domainFormPanel.fieldNames();
+        Map<String, PropertyDescriptor> exportedFields = getFieldsFromExportFile(exportFile);
+
+        // confirm that the ones exported match the ones on the server
+        for (PropertyDescriptor createdField : createdFields)  // created fields will include Key, which wasn't passed in but we do want to validate
+        {
+            String fieldName = createdField.getName();
+
+            if (!fieldName.equals("Key"))
+            {
+                // compare the field definition passed in to the one that came back
+                PropertyDescriptor intendedField = fields.stream().filter(a -> a.getName().equals(fieldName)).findFirst().orElse(null);
+                assertNotNull("a field that wasn't specified [" + fieldName + "] was created", intendedField);
+                verifyExpectedFieldProperties(intendedField, createdField);
+            }
+            // compare the created field properties to the exported field properties
+            PropertyDescriptor exportedField = exportedFields.get(fieldName);
+            assertNotNull("expected field [" + fieldName + "] was not exported", exportedField);
+            verifyExpectedFieldProperties(createdField, exportedField);
+        }
+
+        // now create a list with the set of created Keys
+        String roundTripList = "roundTripListNameCreatedFromJsonFields";
+        EditListDefinitionPage listDefPage = _listHelper.beginCreateList(getProjectName(), roundTripList);
+        listDefPage.getFieldsPanel()
+                .setInferFieldFile(exportFile);
+        listDefPage.selectKeyField("Key");
+        listDefPage.clickSave();
+
+        domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", roundTripList);
+        assertThat("expect list created from export to have the same fields",
+                domainDesignerPage.fieldsPanel().fieldNames(), is(uiFields));
```

Comment:
any reason to also check the field properties here? or do we think that the field names check is sufficient?

Reply:
oh nevermind, that is already being done in testFileImportFieldsToNewSampleType